exp/bufferpool/pool.go

```diff
@@ -4,49 +4,151 @@ package bufferpool
 import "sync"
 
 const (
-	nBuckets = 20
+	defaultMinSize     = 1024
+	defaultBucketCount = 10
 )
 
-// A default global pool.
-var Default Pool
+var defaultPool = &Pool{}
 
-// A pool of buffers, in variable sizes.
+// Set the global pool.  This can be used to override the default implementation
+// to optimize for different allocation profiles. For example, applications that
+// target embedded platforms may wish to lower the minimum buffer size from 1KiB
+// to 512B, or less.
+//
+// The default implementation uses BasicPool, which is suitable for a wide range
+// of applications.  Most applications will not benefit from using a custom pool,
+// so be sure to profile.
+//
+// Set panics if p == nil.
+func Set(p *Pool) {
+	if p == nil {
+		panic("Pool(nil)")
+	}
+
+	defaultPool = p
+}
+
+// Get calls .Get() on the global pool, and resizes the resulting buf
+// to size.
+func Get(size int) []byte {
+	return defaultPool.Get(size)[:size]
+}
+
+// Put calls .Put(buff) on the global pool after zeroing the supplied
```

Comment:
I really think we should move the zeroing logic back into Pool.Put, *especially* now that the interface is gone. Same thing with the capacity check.

Reply:
If we're not splitting the zeroing vs pooling logic, then I see no reason to make the default pool a private variable.  I'll revert back to what we had before and remove the top-level `Get` and `Put` functions.

Reply:
Done.